src/routes/expensesRouter.js

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable function-paren-newline */
+const { Router } = require('express');
+const { mockUsers } = require('./usersRouter.js');
+const expensesRouter = Router();
+const mockExpenses = new Map();
+
+const filterFunctions = {
+  userId: (expenses, userId) =>
+    expenses.filter((expense) => expense.userId === Number(userId)),
+  categories: (expenses, categories) => {
+    const categoryList = categories
+      .split(',')
+      .map((category) => category.trim());
+
+    return expenses.filter((expense) =>
+      categoryList.includes(expense.category),
+    );
+  },
+  dateRange: (expenses, from, to) =>
+    expenses.filter((expense) => {
+      const expenseDate = new Date(expense.spentAt);
+      const startDate = new Date(from);
+      const endDate = new Date(to);
+
+      return expenseDate >= startDate && expenseDate <= endDate;
+    }),
+};
+
+expensesRouter.get('/', (req, res) => {
+  const { userId, from, to, categories } = req.query;
+
+  let filteredExpenses = [...mockExpenses.values()];
+
+  if (userId) {
+    filteredExpenses = filterFunctions.userId(filteredExpenses, userId);
+  }
+
+  if (categories) {
+    filteredExpenses = filterFunctions.categories(filteredExpenses, categories);
+  }
+
+  if (from && to) {
```

Comment:
Again, this filtering process could be inside your _Service_ file.
Aaaand, there is a better approach for applying those filters. If they are already stored in a Hashmap (_filterFunctions_) you could use a single filter to apply then all, no _if_ statements needed.